libpromises/policy.c

```diff
@@ -2627,6 +2702,63 @@ static bool CheckScalarNotEmptyVarRef(const char *scalar)
     return (strcmp("$()", scalar) != 0) && (strcmp("${}", scalar) != 0);
 }
 
+static bool ValidateCustomPromise(const Promise *pp, Seq *errors)
+{
+    assert(pp != NULL);
+
+    const char *const promise_type = pp->parent_section->promise_type;
+    bool valid = true;
+    Seq *attributes = pp->conlist;
+    const size_t length = SeqLength(attributes);
+    for (size_t i = 0; i < length; ++i)
+    {
+        Constraint *attribute = SeqAt(attributes, i);
+        const char *name = attribute->lval;
+        if (StringEqual(name, "action")
+            || StringEqual(name, "action_policy")
+            || StringEqual(name, "ifelapsed")
+            || StringEqual(name, "expireafter")
+            || StringEqual(name, "log_level")
+            || StringEqual(name, "classes")
+            || StringEqual(name, "comment")
+            || StringEqual(name, "depends_on")
+            || StringEqual(name, "handle")
+            || StringEqual(name, "if")
+            || StringEqual(name, "ifvarclass")
+            || StringEqual(name, "meta")
+            || StringEqual(name, "unless")
+            || StringEqual(name, "with"))
+        {
+            // TODO: Remove 1 attribute at a time, test and fix.
+            //       https://tracker.mender.io/browse/CFE-3392
+            SeqAppend(
+                errors,
+                PolicyErrorNew(
+                    POLICY_ELEMENT_TYPE_PROMISE,
+                    pp,
+                    POLICY_ERROR_PROMISE_ATTRIBUTE_NOT_IMPLEMENTED,
+                    name,
+                    promise_type));
```

Comment:
I thought the above attributes were actually the ones that **were** implemented.

Reply:
Nope. These are all the attributes we need to implement / add tests for to ensure they work. Error is added here so they are reserved, i.e. you cannot use them yet.